### DISCUSSION TITLE
Session store not set on request

### DISCUSSION THREAD

**Discussion by init-database** [laravel/framework#40756] [Q&A]
Please give us clear cut explanation how to solve this issue:

especially for API + SPA user:

when we want to login we get this error: "Session store not set on request"

I am kind of skeptic with those answers in stackoverflow thats why I ask here

**Comment by init-database** [comment]
where are you guys????

**Comment by init-database** [comment]
I need help please

**Comment by init-database** [comment]
anyone help

**Reply by ousid** [reply]
Have you looked [here](https://laracasts.com/discuss/channels/laravel/sanctum-throws-session-store-not-set-on-request)?

**Reply by init-database** [reply]
yes, and all the other google search results

**Comment by ousid** [comment]
What stack/package are you using? What do you want to achieve here? Any example code?

**Reply by init-database** [reply]
PHP, Laravel, xampp, mysql, react, material ui, axios, redux tool kit, react router dom v6. I just want to implement user login, I already done all instructions in the docs, but it return that error.

yesterday I add these two line of code inside protected $middleware array in app/Http/Kernel.php file:

\Illuminate\Session\Middleware\StartSession::class,
\Illuminate\View\Middleware\ShareErrorsFromSession::class,

that solve the error but...
is that the correct solution to this problem? is there any security vulnerability I will face in the future?

And why on earth all kind of solutions that spread accross the web not a single one mentioned in official docs?

**Reply by ousid** [reply]
I'm glade that you solved the issue.

But, I really don't get it, Because your question is a little shady, to get a proper response from the community, you need at least to provide the following:

- app versions (e.g: Laravel, React, MySQL, ....)
- Steps to produce
- The issue
- What you are trying to do exactly
- Your code example (what you did exactly in your app).

With all due respect, don't assume that people know what to do, or anyone had the same issue as you are.

**Comment by amitbk** [comment]
If you are using: **$request->session()** try using: **session()** helper directly.
This might solve the problem.

**Reply by hanyfarag7** [reply]
thanks this worked for me could you please explain why this solution instead of $request->session()?

**Reply by amitbk** [reply]
@hanyfarag7 

In Laravel, the session() helper function is a shortcut to accessing the session instance. The session instance is globally available throughout the application, and it's typically initialized during the bootstrapping process.

When you use $request->session(), you're trying to access the session through the request object. However, there might be scenarios where the session isn't properly initialized or set up when the request object is created, leading to the error message "Session store not set on request."

Using the session() helper directly bypasses the request object and directly accesses the session instance. This can sometimes resolve issues related to session initialization because the session() helper function doesn't rely on the request object.

So, if you encounter the "Session store not set on request" error, using session() helper directly can be a workaround to access the session instance without relying on the request object, potentially resolving the issue. However, it's essential to ensure that your session setup is correct and that the error isn't indicative of a deeper issue in your application configuration.
